src/createServer.js

```
@@ -0,0 +1,26 @@
+import express from 'express';
+import cors from 'cors';
+import { userRouter } from './routes/userRouter.js';
+import { authRouter } from './routes/authRouter.js';
+import { profileRouter } from './routes/profileRouter.js';
+import { dbInit } from './utils/dbInit.js';
+import cookieParser from 'cookie-parser';
+
+export function createServer() {
+  const server = express();
+
+  server.use(express.json());
+  server.use(cookieParser());
+  server.use(cors());
+
+  server.get('/', (req, res) => {
+    res.status(200).send('Server is running');
+  });
+
+  server.use('/users', userRouter);
+  server.use('/auth', authRouter);
+  server.use('/profile', profileRouter);
+
+  return server;
```

There is no 404 handler registered. The task requires returning 404 for all routes/pages not defined by the application; add a catch-all handler after the route registrations (before returning the server) such as:

```
server.use((req, res) => res.status(404).json({ message: 'Not Found' }));
```

This ensures unknown routes return the required 404 response as specified in the project description .